### PR TITLE
feat: add wysiwyg task plugin

### PR DESCRIPTION
**Pull request by seonim-ryu**
### Please check if the PR fulfills these requirements
- [x] It's the right issue type on the title
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has a description of the breaking change

### Add WYSIWYG task marker plugin

Add a plugin so that the state changes when the marker of the WYSIWYG task is clicked.

wysiwyg-task

## TODO

* [ ] Move to `dom.ts` and change file name to `dom.js` -> `dom-old.js`

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

**Review by lja1018 — APPROVED** [review]
리뷰완료
고생하셨습니다!

File: apps/editor/src/wysiwyg/helper/dom.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// @TODO move to utils/dom.ts
```
Comment (js87zz) [inline apps/editor/src/wysiwyg/helper/dom.ts:1]: 미리 다 옮겨두는게 좋을 것 같아요ㅎㅎ

File: apps/editor/src/wysiwyg/plugins/taskPlugin.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+            if (!listItem || !isPositionInBox(style, offsetX, offsetY)) {
+              return true;
+            }
+
+            ev.preventDefault();
+
+            const offset = currentPos.before(listItem.depth);
+            const { attrs } = listItem.node;
+            const { checked } = attrs;
```
Comment (js87zz) [inline apps/editor/src/wysiwyg/plugins/taskPlugin.ts:32]: 따로 선언 안하고 써도 괜찮을 것 같아요~

**Review by js87zz — APPROVED** [review]
리뷰완료입니다~! 고생하셨습니다!

File: apps/editor/src/wysiwyg/helper/dom.ts (hunk 1 above)
Comment (seonim-ryu) [reply apps/editor/src/wysiwyg/helper/dom.ts:1]: 이건 커스텀 문법 작업건 머지되면 이후에 전체적으로 파일 수정하는거로 진행할게요~

**Pull request merged**